src/main/java/com/intentfilter/androidpermissions/NotificationSettings.java

```diff
@@ -60,6 +71,11 @@ public Builder withSmallIcon(@DrawableRes int smallIconResId) {
             return this;
         }
 
+        public Builder withColor(@ColorInt int colorResId) {
+            this.notificationSettings.color = colorResId;
```

Comment:
Since we are accepting `@ColorInt` as a param, the name `colorResId` would be inappropriate here. Now, I also think that maybe having a @ColorRes and passing that around might make more sense here, both to keep consistency with the other fields and also because that will allow the developers to pass around any color resource that they might have defined for their theme. What do you think?

Reply:
@nishkarsh ah good catch, you are correct that the name should reflect the type in this case.

Good question about passing a @ColorInt or @ColorRes. The Notification.Builder.setColor() takes a @ColorInt which is why I chose that pattern. I agree it's strange that the Builder takes resource IDs for other things like icons but a direct color in this case. 

We do have access to the Context so we could theoretically de-reference the resource id into a color, but I recommend we stick to mirroring the underlying UI so that developers who are familiar with it won't get confused. 

What do you think? Happy to go either way.

Reply:
I updated the PR to change `colorResId` in this case to `color`. If you'd like us to use a @ColorRes everywhere and dereference it when we build the notification let me know and I can make that change.

Reply:
Thanks for updating the PR. I believe that the color being set in the notification would most probably be a theme color, for which the good practice would be to define in the `colors.xml`. If we could encourage that practice by accepting only a color resource, I would favour that.

Reply:
Ok, I updated the PR to use @ColorRes instead of @ColorInt.